src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -139,34 +34,82 @@ def read_slr_data_from_files(slr_data_dir: pathlib.Path) -> gpd.GeoDataFrame:
     FileNotFoundError
         If the sea level rise data directory does not exist or if there are no CSV files in the specified directory.
     """
-    # Check if the sea level rise data directory exists
-    if not slr_data_dir.exists():
-        raise FileNotFoundError(f"'sea_level_rise' data directory not found: '{slr_data_dir}'.")
-    # Check if there are any CSV files in the specified directory
-    if not any(slr_data_dir.glob("*.csv")):
-        raise FileNotFoundError(f"'sea_level_rise' data files not found in: '{slr_data_dir}'")
-    # Create an empty list to store the sea level rise datasets
-    slr_nz_list = []
-    # Loop through each CSV file in the specified directory
-    for file_path in slr_data_dir.glob("*.csv"):
-        # Read the CSV file into a pandas DataFrame using pyarrow
-        slr_region = csv.read_csv(file_path).to_pandas()
-        # Extract the region name from the file name and add it as a new column in the DataFrame
-        file_name = file_path.stem
-        start_index = file_name.find('projections_') + len('projections_')
-        end_index = file_name.find('_region')
-        region_name = file_name[start_index:end_index]
-        slr_region['region'] = region_name
-        # Append the DataFrame to the list
-        slr_nz_list.append(slr_region)
-        # Log that the file has been successfully loaded
-        log.info(f"Successfully loaded the '{file_path.name}' data file.")
-    # Concatenate all the dataframes in the list and add geometry column
-    slr_nz = pd.concat(slr_nz_list, axis=0).reset_index(drop=True)
-    geometry = gpd.points_from_xy(slr_nz['lon'], slr_nz['lat'], crs=4326)
-    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz, geometry=geometry)
-    # Convert all column names to lowercase
-    slr_nz_with_geom.columns = slr_nz_with_geom.columns.str.lower()
+    # Get the url where sea level rise files need downloading
+    url = 'https://zenodo.org/records/11398538/export/json'
+
+    # Create dictionary to store the sea level rise dataset
+    slr_nz_dict = {}
+    # Request export information json from Zenodo
+    response = requests.get(url)
+    response.raise_for_status()
+    export_json = response.json()
+    # Get necessary links from json file
+    for file_name, file_info in export_json['files']['entries'].items():
+        # Get file name without extension
+        file_name_without_extension = os.path.splitext(file_name)[0]
```

Comment:
```suggestion
        file_name_without_extension = pathlib.Path(file_name).stem
```
We tend to use pathlib in this codebase, since it's more declarative around what it is doing.
Try changing the code to use this kind of format

Reply:
I agree with @LukeParky. Consider using the `pathlib` library instead.